### PR TITLE
Feature/cloud 1389 enable sonar

### PR DESCRIPTION
**Pull request by naveen-variant**
# Description

Adding ability to create sonar project.

Fixes [#24](https://usxtech.atlassian.net/browse/CLOUD-1389)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

```yaml
Test Case 1 [Test on python projects]:
- Steps
  - Go to any demo app and create a branch to the actions-collection branch or use the branch i created.
  - Update the workflow file to point to actions-python branch of enable sonar.
  - Run the workflow. 
- Result
 - See project is created in sonarcloud console with the run results pushed.
 - sample run https://github.com/variant-inc/demo-python-flask-variant-api/actions/runs/1741915207
Test Case 2 [Negative scenario]:
- Steps
  - Check if project already exists in sonar cloud ui and try to rerun the build.
- Result
 - It should not create one as it already exists.
```

 
- [ ] Unit Tests
- [x] Sanity Testing

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -60,6 +60,7 @@ Refer [actions-setup](https://github.com/variant-inc/actions-setup) for document
        with:
          dockerfile_dir_path: '.'
          ecr_repository: naveen-demo-app/demo-repo
+         github_token: ${{ secrets.GITHUB_TOKEN }}
```
Comment (LeeHughsVariant) [inline README.md:63]: With this change, we will probably need to make a post in cloudops-announcements that this is a potentially breaking change for some repos, and they will need to to include the github_token when we update the tag.

Comment (salyateem13) [reply README.md:63]: Can this not be added as an environment variable instead of having each workflow pass it in as input?

Comment (rchengvariant) [reply README.md:63]: @salyateem13 I think Vibin requested it to be an input since he also asked for this in actions-cpp

Comment (naveen-variant) [reply README.md:63]: yes we discussed to have input.

**Comment by salyateem13** [comment]
@naveen-variant tested with actions-collection. Waiting for that PR to be merged to approve

File: action.yml [hunk 2]
```diff
@@ -16,6 +16,12 @@ inputs:
     description: "Framework for Tests"
     required: false
     default: "pytest"
+  github_token:
+    description: "Github token"
+    required: false
```
Comment (rchengvariant) [inline action.yml:21]: This should be true, and no `default`

Comment (naveen-variant) [reply action.yml]: updated.

File: action.yml [hunk 3]
```diff
@@ -16,6 +16,12 @@ inputs:
     description: "Framework for Tests"
     required: false
     default: "pytest"
+  github_token:
+    description: "Github token"
+    required: true
+    default: "dummy-token"
```
Comment (rchengvariant) [inline action.yml:22]: Github Token should not have a default similar to ecr_repository

Comment (naveen-variant) [reply action.yml]: done

**Review by rchengvariant — APPROVED** [review]

**Pull request merged**